File: backend/app/models/company.py
from typing import Any, List, Optional
from pydantic import BaseModel, Field

from app.schemas.company import CompanyEmployeeSummary, CompanyPostSummary
from app.utils.helpers import utc_now_iso


class CompanyModel(BaseModel):
    id: Optional[str] = None
    slug: str
    name: str
    tagline: str
    logoInitials: str = "CX"
    logoGradient: str = "from-cyan-500 to-blue-600"
    industry: str
    size: str
    headquarters: str
    foundedYear: str
    fundingStage: str
    websiteUrl: str
    about: str
    mission: str
    techStack: List[str] = Field(default_factory=list)
    benefits: List[str] = Field(default_factory=list)
    openJobsCount: int = 0
    followers: List[str] = Field(default_factory=list)  # userIds
    employees: List[CompanyEmployeeSummary] = Field(default_factory=list)
    posts: List[CompanyPostSummary] = Field(default_factory=list)
    jobs: List[Any] = Field(default_factory=list)
    createdAt: str = Field(default_factory=utc_now_iso)
    updatedAt: Optional[str] = None

File: src/types/company.ts
import { JobItem } from './index';

export interface CompanyPost {
  id: string;
  title: string;
  date: string;
  content: string;
  author: string;
  authorRole: string;
  likesCount: number;
}

export interface CompanyEmployee {
  id: string;
  name: string;
  role: string;
  avatarInitials: string;
  avatarGradient: string;
  isConnected: boolean;
}

export interface CompanyProfile {
  id: string;
  slug: string;
  name: string;
  tagline: string;
  logoInitials: string;
  logoGradient: string;
  industry: string;
  size: string;
  headquarters: string;
  founded: string;
  funding: string;
  website: string;
  isFollowing: boolean;
  followersCount: number;
  about: string;
  mission: string;
  techStack: string[];
  benefits: string[];
  openJobsCount: number;
  jobs: JobItem[];
  posts: CompanyPost[];
  employees: CompanyEmployee[];
}
